apps/web/pages/settings/my-account/profile.tsx

```diff
@@ -180,128 +179,135 @@ const ProfileView = () => {
 
   const defaultValues = {
-    username: user.username || "",
-    avatar: avatar.avatar || "",
-    name: user.name || "",
-    email: user.email || "",
-    bio: user.bio || "",
+    username: user?.username || "",
+    avatar: avatar?.avatar || "",
+    name: user?.name || "",
+    email: user?.email || "",
+    bio: user?.bio || "",
   };
 
   return (
     <>
       <Meta title={t("profile")} description={t("profile_description", { appName: APP_NAME })} />
-      <ProfileForm
-        key={JSON.stringify(defaultValues)}
-        defaultValues={defaultValues}
-        onSubmit={(values) => {
-          if (values.email !== user.email && isCALIdentityProviver) {
-            setTempFormValues(values);
-            setConfirmPasswordOpen(true);
-          } else {
-            mutation.mutate(values);
-          }
-        }}
-        extraField={
-          <div className="mt-8">
-            <UsernameAvailabilityField
-              user={user}
-              onSuccessMutation={async () => {
-                showToast(t("settings_updated_successfully"), "success");
-                await utils.viewer.me.invalidate();
-              }}
-              onErrorMutation={() => {
-                showToast(t("error_updating_settings"), "error");
-              }}
-            />
-          </div>
-        }
-      />
-
-      <hr className="my-6  border-neutral-200" />
-
-      <Label>{t("danger_zone")}</Label>
-      {/* Delete account Dialog */}
-      <Dialog open={deleteAccountOpen} onOpenChange={setDeleteAccountOpen}>
-        <DialogTrigger asChild>
-          <Button
-            data-testid="delete-account"
-            color="destructive"
-            className="mt-1 border-2"
-            StartIcon={Icon.FiTrash2}>
-            {t("delete_account")}
-          </Button>
-        </DialogTrigger>
-        <DialogContent
-          title={t("delete_account_modal_title")}
-          description={t("confirm_delete_account_modal", { appName: APP_NAME })}
-          type="creation"
-          Icon={Icon.FiAlertTriangle}>
-          <>
-            <p className="mb-7">{t("delete_account_confirmation_message", { appName: APP_NAME })}</p>
-            {isCALIdentityProviver && (
-              <PasswordField
-                data-testid="password"
-                name="password"
-                id="password"
-                autoComplete="current-password"
-                required
-                label="Password"
-                ref={passwordRef}
-              />
-            )}
-
-            {user?.twoFactorEnabled && isCALIdentityProviver && (
-              <Form handleSubmit={onConfirm} className="pb-4" form={form}>
-                <TwoFactor center={false} />
-              </Form>
-            )}
-
-            {hasDeleteErrors && <Alert severity="error" title={deleteErrorMessage} />}
-            <DialogFooter>
+      {isPageLoading ? (
+        <SkeletonLoader />
+      ) : (
```

Comment:
- This condition should run in a container component
- ProfileForm should receive full `defaultValues` on the first render. If not there will be inconsistencies.